### PR TITLE
Improve `posix-futex` debugging

### PR DESCRIPTION
**Pull request by skuenzer**
### Base target

 - Architecture(s): [all]
 - Platform(s): [all]
 - Application(s): [N/A]

### Additional configuration

 - `CONFIG_LIBPOSIX_FUTEX=y`
 - `CONFIG_LIBPOSIX_FUTEX_DEBUG=y`

### Description of changes

This PR adds a set of debug messages to better understand the futex state, like threads getting into blocked state or waking up because of timeouts.

### CODE REVIEW

**Comment by razvand** [comment]
@skuenzer , could you please fix the `checkpatch` issues?

**Comment by razvand** [comment]
> @skuenzer , could you please fix the `checkpatch` issues?

Apart from the `checkpatch` issues, everything works OK. After fixing the checkpatch issues and after @maniatro111 and @csvancea provide their reviews, I will do the final approval of the PR.

File: lib/posix-futex/futex.c [hunk 1]
```diff
@@ -101,50 +102,57 @@ static int futex_wait(uint32_t *uaddr, uint32_t val, const __nsec *timeout)
...
+	uk_list_add_tail(&f.list_node, &futex_list);
+	uk_spin_unlock(&futex_list_lock);
+	ukplat_lcpu_restore_irqf(irqf);
 
-		uk_sched_yield();
+	if (timeout) {
+		/* Block at most until `timeout` nanosecs */
+		uk_pr_debug("FUTEX_WAIT: Wait %"__PRIsnsec" nsec for wake-up\n",
+				(__snsec) (*timeout));
+		uk_thread_block_until(current, (__snsec) (*timeout));
```
Comment (csvancea) [inline lib/posix-futex/futex.c:126]: Minor: maybe turn the expression `(__snsec) (*timeout)` into a variable since it is used multiple times?

**Review by csvancea — APPROVED** [review]
Apart from the checkpatch and that minor nitpick -- LGTM

Reviewed-by: Cosmin Vancea csvancea@gmail.com

**Comment by maniatro111** [comment]
@skuenzer Looks good to me. One small preference. Either add a space before and after the equal sign or remove them from both sides so that the result will be `Condition = 2147483664` instead of `Condition =2147483664`.

Reviewed-by: Florin Postolache <florin.postolache.of@gmail.com>

**Review by flpostolache — COMMENTED** [review]
@skuenzer Looks good to me. One small preference. Either add a space before and after the equal sign or remove them from both sides so that the result will be `Condition = 2147483664` instead of `Condition =2147483664`.

**Comment by skuenzer** [comment]
> @skuenzer , could you please fix the `checkpatch` issues?

As far as I can see, these are our standard warnings for long strings. Our `checkpatch` clone does not yet support detecting that we are having a print command with `uk_pr_*` that is why it is warning.

File: lib/posix-futex/futex.c (hunk 1 above)
Comment (skuenzer) [reply lib/posix-futex/futex.c:126]: With this patch, it is only used a second time in the debug message that I add. However, normally this message is not compiled in, so I find adding a variable that stores the dereferenced value of `timeout` for a single occurrence unnecessary.

**Comment by skuenzer** [comment]
> @skuenzer Looks good to me. One small preference. Either add a space before and after the equal sign or remove them from both sides so that the result will be `Condition = 2147483664` instead of `Condition =2147483664`.

Right, sorry, I did not notice it. For sure I am fixing this.

**Review by flpostolache — APPROVED** [review]
Looks good now. All checkpatch issues are caused by `uk_pr_*` commands which are not ignored at the moment.

Reviewed-by: Florin Postolache <florin.postolache.of@gmail.com>

**Review by razvand — APPROVED** [review]
Approved-by: Razvan Deaconescu <razvand@unikraft.io>

**Pull request closed without merging**